doc/lsst.verify/tasks/lsst.verify.compatibility.MetricsControllerTask.rst

```diff
@@ -0,0 +1,105 @@
+The datasets that ``MetricsControllerTask`` consumes depend on the :lsst-task:`~lsst.verify.compatibility.MetricTask`\ s to be run, and are handled automatically.
+
+``MetricsControllerTask`` is not a command-line task, but may be called from within both task- and non-task pipelines.
+
+.. _lsst.verify.compatibility.MetricsControllerTask-summary:
+
+Processing summary
+==================
+
+Unlike most tasks, ``MetricsControllerTask`` has a `~MetricsControllerTask.runDataRefs` method that takes a list of data references.
+``MetricsControllerTask`` calls every :lsst-task:`~lsst.verify.compatibility.MetricTask` in :lsst-config-field:`~lsst.verify.compatibility.MetricsControllerConfig.measurers` on every data reference, loading any datasets necessary.
+It produces one `~lsst.verify.Job` object for each input data reference, and writes them to disk.
+
+.. _lsst.verify.compatibility.MetricsControllerTask-api:
+
+Python API summary
+==================
+
+.. lsst-task-api-summary:: lsst.verify.compatibility.MetricsControllerTask
+
+.. _lsst.verify.compatibility.MetricsControllerTask-subtasks:
+
+Retargetable subtasks
+=====================
+
+.. lsst-task-config-subtasks:: lsst.verify.compatibility.MetricsControllerTask
+
+.. _lsst.verify.compatibility.MetricsControllerTask-configs:
+
+Configuration fields
+====================
+
+.. lsst-task-config-fields:: lsst.verify.compatibility.MetricsControllerTask
+
+.. _lsst.verify.compatibility.MetricsControllerTask-indepth:
+
+In Depth
+========
+
+Because ``MetricsControllerTask`` applies every :lsst-task:`~lsst.verify.compatibility.MetricTask` to every input data reference indiscriminately, it may not give good results with metrics or data references having a mixture of granularities (e.g., CCD-level, visit-level, dataset-level).
+The recommended way around this limitation is to create multiple ``MetricsControllerTask`` objects, and configure each one for metrics of a single granularity.
+
+Each :lsst-task:`~lsst.verify.compatibility.MetricTask` in a ``MetricsControllerTask`` must measure a different metric, or they will overwrite each others' values.
+
+.. _lsst.verify.compatibility.MetricsControllerTask-examples:
+
+Examples
+========
+
+Typically, a user of ``MetricsControllerTask`` will configure it with tasks that have `register` decorator:
+
+.. code-block:: py
+
+   from lsst.verify.compatibility import register, MetricTask, MetricsControllerTask
+
+   @register("spherical")
+   class SphericityMetric(MetricTask):
+       ...
+
+   @register("cow")
+   class BovinityMetric(MetricTask):
+       ...
+
+   config = MetricsControllerTask.ConfigClass()
+   config.measurers = ["spherical", "cow"]
+   config.measurers["spherical"].answer = 42
+   task = MetricsControllerTask(config)
+
+   # CCD-level metrics need CCD-level datarefs
+   # Exact dataset type doesn't matter
+   datarefs = [butler.subset("calexp", visit=42, ccd=ccd)
+       for ccd in range(1, 101)]
```

Comment:
oof... @jonathansick ?

Reply:
I'm okay with the line break but I don't like the indenting. What's wrong with:
```python
datarefs = [butler.subset("calexp", visit=42, ccd=ccd)
            for ccd in range(1, 101)]
```
? (ie the `for` under the `but`).

Reply:
@timj no objections. In hindsight, maybe I should have written the code in a Python file first so that it could be flake8-checked...

Reply:
@parejkoj Sorry, what's the issue here? The sample @timj produced looks great.

Reply:
If you folks don't want to code to 79 characters, you can't totally change the setup.cfg to the DM style. The setup.cfg here simply predates lsst.verify's integration with the DM Stack when it was a SQuaRE product alone and we were just coding to PEP 8.

Reply:
The very short line width in doc builds is not great. Is there anything that could be done about that, or do we need an even shorter line length requirement for code examples in docs?

Reply:
Don't worry about that.

Reply:
For the record, I *don't* support such a requirement. For one thing, the display of the doc builds seems to be slightly browser-dependent (or maybe available-font-dependent), so any hard limit would need to be carefully researched. For another, having long lines and letting the user scroll horizontally sometimes looks better; I opted not to in this case because it was just the one line.

Reply:
All I'm saying is that the theme you see the docs rendered in now is not the theme we're going into production with. tl;dr, produce documentation content and trust that as long as your code samples are formatted neatly, they'll look great in the "final" version of the documentation. This PR is not the best time or place to talk about documentation engineering.

Reply:
Sounds good!